src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -339,154 +347,127 @@ DatabaseShardImp::importShard(
     std::uint32_t shardIndex,
     boost::filesystem::path const& srcDir)
 {
-    using namespace boost::filesystem;
-    try
-    {
-        if (!is_directory(srcDir) || is_empty(srcDir))
-        {
-            JLOG(j_.error()) << "invalid source directory " << srcDir.string();
-            return false;
-        }
-    }
-    catch (std::exception const& e)
-    {
-        JLOG(j_.error()) << "exception " << e.what() << " in function "
-                         << __func__;
+    auto fail = [j = j_, shardIndex](std::string const& msg) {
```

Comment:
Should this fail lambda also remove the index from `preparedIndexes_`? If we don't remove the failed shard's index, it seems like the user won't be able to retry the download until the server restarts, since the `ShardArchiveHandler`'s call to `DatabaseShardImp::prepareShard` will fail.